src/ADC.c

```diff
@@ -187,187 +195,40 @@ extern void set_motor_current_temp_MUX(ADC_sensors_t ADC_sensor_x) {
-			Read_Position = 15;
-			ADC_count = 0;
-			ADC_SoftwareStartConv(ADC1);
-			break;
-		//read pressure
-		case Water_ADC:
-			Read_Position = 16;
-			ADC_count = 0;
-			ADC_SoftwareStartConv(ADC1);
-			break;
-		}
+	Read_Position = ADC_sensor_x;
+	ADC_count = 0;
+	ADC_SoftwareStartConv(ADC1);
 }
 
 extern uint16_t return_ADC_value(ADC_sensors_t ADC_sensor_x){
-	switch (ADC_sensor_x) {
-	//read current
-		case Curr_ADC1:
-			Read_Position = 0;
-			average = calculate_average(Read_Position)*1.79;
-			return double_to_int(average);
-			break;
-		case Curr_ADC2:
-			Read_Position = 1;
-			average = calculate_average(Read_Position)*1.79;
-			return double_to_int(average);
-			break;
-		case Curr_ADC3:
-			Read_Position = 2;
-			average = calculate_average(Read_Position)*1.79;
-			return double_to_int(average);
-			break;
-		case Curr_ADC4:
-			Read_Position = 3;
-			average = calculate_average(Read_Position)*1.79;
-			return double_to_int(average);
-			break;
-		case Curr_ADC5:
-			Read_Position = 4;
-			average = calculate_average(Read_Position)*1.79;
-			return double_to_int(average);
-			break;
-		case Curr_ADC6:
-			Read_Position = 5;
-			average = calculate_average(Read_Position)*1.79;
-			return double_to_int(average);
-			break;
-		case Curr_ADC7:
-			Read_Position = 6;
-			average = calculate_average(Read_Position)*1.79;
-			return double_to_int(average);
-			break;
-		case Curr_ADC8:
-			Read_Position = 7;
-			average = calculate_average(Read_Position)*1.79;
-			return double_to_int(average);
-			break;
-		//read Temperature
-		case Temp_ADC1:
-			Read_Position = 8;
-			return calculate_average(Read_Position)*0.0177;
-			break;
-		case Temp_ADC2:
-			Read_Position = 9;
-			return calculate_average(Read_Position)*0.0177;
-			break;
-		case Temp_ADC3:
-			Read_Position = 10;
-			return calculate_average(Read_Position)*0.0177;
-			break;
-		case Temp_ADC4:
-			Read_Position = 11;
-			return calculate_average(Read_Position)*0.0177;
-			break;
-		case Temp_ADC5:
-			Read_Position = 12;
-			return calculate_average(Read_Position)*0.0177;
-			break;
-		case Temp_ADC6:
-			Read_Position = 13;
-			return calculate_average(Read_Position)*0.0177;
-			break;
-		case Temp_ADC7:
-			Read_Position = 14;
-			return calculate_average(Read_Position)*0.0177;
-			break;
-		case Temp_ADC8:
-			Read_Position = 15;
-			return calculate_average(Read_Position)*0.0177;
-			break;
-		//read pressure
-		case Water_ADC:
-			Read_Position = 16;
-			return calculate_average(Read_Position)*0.088;
-			break;
-		}
+	if(ADC_sensor_x <= Curr_ADC8){
+		average= calculate_average(ADC_sensor_x)*1.79;
+		return double_to_int(average);
+	} else if((ADC_sensor_x >= Temp_ADC1) && (ADC_sensor_x <= Temp_ADC8)){
+		average= calculate_average(ADC_sensor_x)*1.79;
+		return double_to_int(average);
+	} else if(ADC_sensor_x == Water_ADC){
+		average = calculate_average(ADC_sensor_x)*0.088;
+		return double_to_int(average);
+	} else return 0;
+}
+
+extern void ADCTask(void *dummy){
```

Comment:
It looks like you have this as extern because you are initializing the task in main.c. You don't have to initialize the task in the vGeneralTaskInit() function. You can call xTaskCreate() anywhere. I would recommend putting it in the ADC init function and remove the extern. Look at the FSM code for an example of this.